src/if.js

```diff
@@ -40,9 +40,34 @@ export class If {
   * @param newValue The new value
   */
   valueChanged(newValue) {
+    if (this.__queuedChanges) {
+      this.__queuedChanges.push(newValue);
+      return;
+    }
+
+    let maybePromise = this._runValueChanged(newValue);
+    if (maybePromise instanceof Promise) {
+      let queuedChanges = this.__queuedChanges = [];
+
+      let runQueuedChanges = () => {
+        if (!queuedChanges.length) {
+          this.__queuedChanges = undefined;
+          return;
+        }
+
+        let nextPromise = this._runValueChanged(queuedChanges.shift()) || Promise.resolve();
+        nextPromise.then(runQueuedChanges);
+      };
+
+      maybePromise.then(runQueuedChanges);
+    }
+  }
+
+  _runValueChanged(newValue) {
     if (!newValue) {
+      let viewOrPromise;
       if (this.view !== null && this.showing) {
-        let viewOrPromise = this.viewSlot.remove(this.view);
+        viewOrPromise = this.viewSlot.remove(this.view);
```

Comment:
@EisenbergEffect  @martingust  I don't think I'll touch this because it's very edgy, but for the sake of discussion: 
There's a race condition when the `if` is unbound while an exit animation is in progress.
💣 Views can be returned to cache before the animation completes and before they are actually remove from the DOM :(
If the cache is not used, there's a possibility that when the `if` is re-bound, users see the exit animation complete, although that's less bad.
BTW: I'm not very familiar with our animation APIs, is there a way to (optionally?) skip the enter animation on first render?